packages/permissions/src/components/authorized/authorized.tsx

```diff
@@ -8,40 +9,29 @@ import {
 } from '../../utils/has-permissions';
 import getDisplayName from '../../utils/get-display-name';
 
-class Authorized extends React.Component {
-  static displayName = 'Authorized';
-  static propTypes = {
-    shouldMatchSomePermissions: PropTypes.bool,
-    /**
-     * This custom prop-types verifies that any permission passed as
-     * `demandedPermissions` actually exists on the `actualPermissions`.
-     *
-     * This was previously achieved through a validation via the constants (`Object.keys(permissions)`).
-     * This was deemded to not be flexible enough to introduce new permissions as they have to be
-     * added to the constants and released each time.
-     */
-    demandedPermissions(props, propName, componentName) {
-      const namesOfNonConfiguredPermissions = getInvalidPermissions(
-        props.demandedPermissions,
-        props.actualPermissions
-      );
-
-      if (namesOfNonConfiguredPermissions.length > 0) {
-        return new Error(
-          `Invalid prop \`${propName}\` supplied to \`${componentName}\`. The permission(s) ${namesOfNonConfiguredPermissions.toString()} is/are not configured through \`actualPermissions\`.`
-        );
-      }
+type RenderProp = (isAuthorized: boolean) => React.ReactNode;
+type Props = {
+  shouldMatchSomePermissions?: boolean;
+  demandedPermissions: TPermissionName[];
+  actualPermissions: TPermissions;
+  render: RenderProp;
+};
```

Comment:
Let’s also add `children?: never`

Reply:
Also, we need to type the default props as well, check out this article

https://medium.com/@martin_hotell/10-typescript-pro-tips-patterns-with-or-without-react-5799488d6680